### PR TITLE
Let Xvfb choose the next free display (fixes #47)

### LINKED ISSUE #47: Graciously handle situation where display :1 is already used
Right now, starting Xvfb fails silently and an error is raised at the end of the build process when trying to kill it.

### PR DESCRIPTION
**Pull request by brechtm**
Oops. I first accidentally pushed this to master which closed #47. I reverted the push to master...

@Modelmat Please review this if you have the time. After, I'll bump to 0.0.12 and make a new release.

P.S. I hope your exams went well?

### CODE REVIEW

File: sphinxcontrib/drawio/__init__.py [hunk 1]
```diff
@@ -248,10 +253,12 @@ def on_build_finished(app: Sphinx, exc: Exception) -> None:
         src = os.path.join(this_file_path, "drawio.css")
         dst = os.path.join(app.outdir, "_static")
         copy_asset(src, dst)
-
-    if is_headless(app.builder.config):
-        os.kill(app.builder.config.xvfb_pid, signal.SIGTERM)
-        os.waitid(os.P_PID, app.builder.config.xvfb_pid, os.WEXITED)
+    if app.config._xvfb:
```
Comment (Modelmat) [inline sphinxcontrib/drawio/__init__.py:256]: Emptyline before the if-block please.

File: sphinxcontrib/drawio/__init__.py [hunk 2]
```diff
@@ -230,16 +225,26 @@ def drawio_export(builder: Builder, options: dict, in_filename: str,
...
-
-        if process.poll() is not None:
-            raise OSError("Failed to start Xvfb process"
-                          "\n[stdout]\n{}\n[stderr]{}".format(*process.communicate()))
-
+        logger.info("running in headless mode, starting Xvfb")
+        with TemporaryFile() as fp:
+            fd = fp.fileno()
+            xvfb = Popen(["Xvfb", "-displayfd", str(fd), "-screen",
+                          "0", "1280x768x16"], pass_fds=(fd, ),
```
Comment (Modelmat) [inline sphinxcontrib/drawio/__init__.py:232]: any reason why this display size was chosen specifically?

**Review by Modelmat — APPROVED** [review]
Looks fine to me.
I haven't tested it locally - and can't quickly - but hopefully, the CI working should indicate that it is working.

**Comment by Modelmat** [comment]
My exams went okay, thanks :)

Did you force push to master? I can't see the revert commit & Github shows the commit which closed #47 as not existing on any branch. If so, please try not to in future - it can create issues when pulling changes.

File: sphinxcontrib/drawio/__init__.py (hunk 1 above)
Comment (brechtm) [reply sphinxcontrib/drawio/__init__.py:257]: Done! Note that was not done consistently in the source code even before Itay's or my commits, so I don't know what the rule is here. Same for _else_ statements. But I guess this won't matter once we'll be using Black.

File: sphinxcontrib/drawio/__init__.py (hunk 2 above)
Comment (brechtm) [reply sphinxcontrib/drawio/__init__.py:232]: I have no idea. Do [you](https://github.com/Modelmat/sphinxcontrib-drawio/commit/8d5aa4b799fef4c41acb2ec14751408eb757a2bf#diff-74ea612f243ff0b44c6c2000a0e8d6ede7599341ce76fdeff8a5ee9aafc1a004R200)? 😉

**Comment by brechtm** [comment]
> My exams went okay, thanks :)

Glad to hear that! 👍 

> Did you force push to master? I can't see the revert commit & Github shows the commit which closed #47 as not existing on any branch. If so, please try not to in future - it can create issues when pulling changes.

Yes, I force pushed just a couple of minutes after I accidentally pushed the PR commit, so the chance of someone already having pulled the new commit was very small.

You can deny force pushes to master in the repository's settings, if you wish.

File: sphinxcontrib/drawio/__init__.py (hunk 2 above)
Comment (Modelmat) [reply sphinxcontrib/drawio/__init__.py:232]: Haha :)
I guess not..

**Pull request merged**